WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -108,59 +112,74 @@ public ValueTask LoadSearchApplicationAsync()
             {
                 if (task.IsCompletedSuccessfully)
                     _ = task.Result?.Query.Pages.ParallelForEachAsync(async entry =>
-                    {
-                        (_, PageView value) = entry;
-                        string resultName = value.Extract;
-                        string displayedName = value.Title;
-                        double score = displayedName.SearchDistanceScore(searchedText);
-                        string pageId = value.PageId.ToString();
-                        string wikiUrl = WikiRootUrl + displayedName;
-                        BitmapImageResult bitmapImageResult;
-
-                        string additionalInfo = "";
-                        if (!string.IsNullOrWhiteSpace(resultName))
                         {
-                            using var reader = new StringReader(resultName);
-                            additionalInfo = await reader.ReadLineAsync().ConfigureAwait(false) ?? resultName;
-                        }
-
-                        if (value.Thumbnail != null)
-                        {
-                            string imgUrl = value.Thumbnail.Source;
-                            using var imageClient = new HttpClient();
-                            Stream stream = await imageClient.GetStreamAsync(imgUrl, cancellationToken)
+                            WikiPreviewSearchResult wikiPreviewSearchResult =
+                                await GenerateSearchResult(entry.Value, searchedText);
+                            await channel.Writer.WriteAsync(wikiPreviewSearchResult, CancellationToken.None)
```

Comment:
You should check that this is not `null` before writing to channel, since you're returning `default` in some cases in `GenerateSearchResult`